lib/spack/spack/cmd/load.py

```diff
@@ -4,22 +4,65 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import argparse
-from spack.cmd.common import print_module_placeholder_help, arguments
+import sys
 
-description = "add package to environment using `module load`"
-section = "modules"
+import llnl.util.tty as tty
+
+import spack.cmd
+from spack.cmd.common import arguments
+import spack.environment as ev
+import spack.util.environment
+import spack.user_environment as uenv
+
+description = "add package to the user environment variables"
+section = "user environment"
 level = "short"
 
 
 def setup_parser(subparser):
     """Parser is only constructed so that this prints a nice help
        message with -h. """
+    shells = subparser.add_mutually_exclusive_group()
+    shells.add_argument(
+        '--sh', action='store_const', dest='shell', const='sh',
+        help="print sh commands to load the package")
+    shells.add_argument(
+        '--csh', action='store_const', dest='shell', const='csh',
+        help="print csh commands to load the package")
+
     subparser.add_argument(
-        'spec', nargs=argparse.REMAINDER,
-        help="spec of package to load with modules "
+        'specs', nargs=argparse.REMAINDER,
+        help="spec of package to load"
     )
     arguments.add_common_arguments(subparser, ['recurse_dependencies'])
 
 
 def load(parser, args):
-    print_module_placeholder_help()
+    env = ev.get_env(args, 'load')
```

Comment:
Yes, I think it is primarily useful for directory-based environments

I've been debating with myself whether activating an environment should add all of its packages to the list of loaded packages. They're not loaded exactly the same way as they would be via `spack load` (they're loaded via the view, not their prefixes), but they are in some way analogous.

Reply:
Let's skip the last part for this initial pass. It might be something we want to unify later, but adding everything in an environment to the loaded packages could explode the env var more than we'll do already.